[1.10] Failed to build on FreeBSD 13.1

* OS: FreeBSD
* OS Version: 13.1
* Architecture: amd64

**Steps to reproduce**

```bash
git clone https://github.com/tarantool/tarantool.git && cd tarantool
git checkout 1.10
git submodule update --recursive --init
cmake . -DCMAKE_BUILD_TYPE=RelWithDebInfo -DENABLE_WERROR=ON
gmake -j
```

**Actual behaviour**

```bash
[  7%] Building C object src/lib/small/CMakeFiles/small.dir/small/small_features.c.o
[  7%] Building C object src/lib/small/CMakeFiles/small.dir/small/slab_cache.c.o
[  7%] Building C object src/lib/small/CMakeFiles/small.dir/small/region.c.o
[  8%] Building C object src/lib/small/CMakeFiles/small.dir/small/mempool.c.o
[  8%] Building C object src/lib/small/CMakeFiles/small.dir/small/slab_arena.c.o
/home/freebsd/tarantool/src/lib/small/small/slab_arena.c:220:9: error: variable 'total' set but not used [-Werror,-Wunused-but-set-variable]
        size_t total = 0;
               ^
1 error generated.
gmake[2]: *** [src/lib/small/CMakeFiles/small.dir/build.make:132: src/lib/small/CMakeFiles/small.dir/small/slab_arena.c.o] Error 1
gmake[1]: *** [CMakeFiles/Makefile2:3553: src/lib/small/CMakeFiles/small.dir/all] Error 2
gmake: *** [Makefile:156: all] Error 2
```

**Expected behaviour**

Build is successful.

After local cherry-picking of https://github.com/tarantool/tarantool/commit/a6b76abdd21792ef56629c61bafff0dc86ca57bb I can see the following error:

```bash
[  7%] Building C object src/lib/small/CMakeFiles/small.dir/small/slab_arena.c.o
/home/freebsd/tarantool/src/lib/small/small/slab_arena.c:220:9: error: variable 'total' set but not used [-Werror,-Wunused-but-set-variable]
        size_t total = 0;
               ^
1 error generated.
gmake[2]: *** [src/lib/small/CMakeFiles/small.dir/build.make:132: src/lib/small/CMakeFiles/small.dir/small/slab_arena.c.o] Error 1
gmake[1]: *** [CMakeFiles/Makefile2:3553: src/lib/small/CMakeFiles/small.dir/all] Error 2
gmake: *** [Makefile:156: all] Error 2
```

Looks like we need to bump version of the `small` submodule.

Cherry-picked commits https://github.com/tarantool/tarantool/commit/760319ea91cac732866a0d8e58e3fe97a83ce431 and  https://github.com/tarantool/tarantool/commit/678e47eec5eb15921667c72f53b8cfa90c857d8d solved the mentioned issues  above but now I can see this:
```bash
[ 49%] Building C object src/box/CMakeFiles/tuple.dir/tuple_update.c.o
/home/freebsd/tarantool/src/box/tuple_update.c:877:11: error: variable 'total_field_count' set but not used [-Werror,-Wunused-but-set-variable]
        uint32_t total_field_count = 0;
                 ^
1 error generated.
gmake[2]: *** [src/box/CMakeFiles/tuple.dir/build.make:104: src/box/CMakeFiles/tuple.dir/tuple_update.c.o] Error 1
gmake[1]: *** [CMakeFiles/Makefile2:4377: src/box/CMakeFiles/tuple.dir/all] Error 2
gmake: *** [Makefile:156: all] Error 2
```

--- a/src/box/tuple_update.c
+++ b/src/box/tuple_update.c
@@ -902,7 +902,7 @@ update_write_tuple(struct tuple_update *update, char *buffer, char *buffer_end)
 		}
 		total_field_count += field_count;
 	}
-
+	(void)total_field_count;
 	assert(rope_size(update->rope) == total_field_count);
 	assert(new_data <= buffer_end);
 	return new_data - buffer; /* real_tuple_size */

